src/shared/nevermined/nvm.service.ts

```
@@ -236,6 +237,37 @@ export class NeverminedService {
     }
   }
 
+  async uploadIPFS(content: Buffer, filename: string): Promise<string> {
+    try {
+      Logger.debug(`Uploading to IPFS ${filename}`);
+      const ipfsAuthToken = this.getIPFSAuthToken();
+
+      const ipfs = IpfsHttpClientLite({
+        apiUrl: this.config.get('IPFS_GATEWAY'),
+        ...(ipfsAuthToken && {
+          headers: { Authorization: `Basic ${ipfsAuthToken}` }
+        })
+      });
+      const addResult = await ipfs.add(content);
+      return `cid://${addResult[0].hash}`;
+    } catch (e) {
+      Logger.error(`Uploading ${filename}: IPFS error ${e}`);
+      throw new InternalServerErrorException(e);
+    }
+  }
+
+  private getIPFSAuthToken(): string | undefined {
+
+    if (!this.config.get('IPFS_PROJECT_ID') || !this.config.get('IPFS_PROJECT_SECRET')) {
+      Logger.warn(`Infura IPFS_PROJECT_ID or IPFS_PROJECT_SECRET are not set - disabling ipfs auth`)      
+      return undefined
```

I think this should just throw an error with a message saying something like `this node does not support upload to ipfs`

---

this warns but allows you still can use a public ipfs now without project_id or secret